cwms-data-api/src/main/java/cwms/cda/api/StreamLocationController.java

```diff
@@ -0,0 +1,279 @@
+            if(UnitSystem.EN.getValue().equals(unitSystem)) {
+                stationUnits = "mi";
+                areaUnits = "mi2";
+                stageUnits = "ft";
+            }
+
+            List<StreamLocation> streamLocations = dao.retrieveStreamLocations(streamId, locationId, stationUnits, stageUnits, areaUnits, office);
+            String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) :
+                    Formats.JSONV1;
+            ContentType contentType = Formats.parseHeader(formatHeader, StreamLocation.class);
+            ctx.contentType(contentType.toString());
+            String serialized = Formats.format(contentType, streamLocations, StreamLocation.class);
+            ctx.result(serialized);
+            ctx.status(HttpServletResponse.SC_OK);
+            requestResultSize.update(serialized.length());
+        }
+    }
+
+    @OpenApi(
+            pathParams = {
+                    @OpenApiParam(name = NAME, required = true, description = "Specifies the location-id of "
+                            + "the stream location to be retrieved."),
+            },
+            queryParams = {
+                    @OpenApiParam(name = OFFICE, required = true, description = "Specifies the owning office of "
+                            + "the stream location to be retrieved."),
+                    @OpenApiParam(name = STREAM_ID, required = true, description = "Specifies the stream-id of the "
+                            + "stream location to be retrieved."),
+                    @OpenApiParam(name = UNIT_SYSTEM, type = UnitSystem.class, description = UnitSystem.DESCRIPTION)
+            },
+            responses = {
+                    @OpenApiResponse(status = STATUS_200,
+                            content = {
+                                    @OpenApiContent(isArray = true, type = Formats.JSONV1, from = StreamLocation.class),
+                                    @OpenApiContent(isArray = true, type = Formats.JSON, from = StreamLocation.class)
+                            })
+            },
+            description = "Returns CWMS Stream Location Data",
+            tags = {TAG}
+    )
+    @Override
+    public void getOne(@NotNull Context ctx, @NotNull String locationId) {
+        String office = requiredParam(ctx, OFFICE);
+        String streamId = requiredParam(ctx, STREAM_ID);
+        try (Timer.Context ignored = markAndTime(GET_ONE)) {
+            DSLContext dsl = getDslContext(ctx);
+            StreamLocationDao dao = new StreamLocationDao(dsl);
+            String unitSystem = queryParamAsClass(ctx, new String[]{UNIT_SYSTEM, },
+                    String.class, UnitSystem.SI.getValue(), metrics,
+                    name(StreamLocationController.class.getName(), GET_ONE));
+            String stationUnits = "km";
+            String areaUnits = "km2";
+            String stageUnits = "m";
+            if(UnitSystem.EN.getValue().equals(unitSystem)) {
```

Comment:
Let's bring up unit-system vs individual units at the CDA dev meeting.

Reply:
Decided to use individual unit parameters during CDA dev meeting